utils/link_retrieval.py

```diff
@@ -1,97 +1,97 @@
-    response_dataframe = pd.DataFrame(columns=['uuid','url'])
-    for obj in response_list:
-        obj_dataframe = pd.DataFrame(obj, index=[0])
-        obj_dataframe.rename(columns={"ytUrl": "url"}, inplace=True)
-        response_dataframe = pd.concat([response_dataframe, obj_dataframe], ignore_index=True)
+        for obj in response_list:
+            obj_dataframe = pd.DataFrame(obj, index=[0])
+            obj_dataframe.rename(columns={"id": "id", "ytUrl": "url", "game": "game"}, inplace=True)
+            response_dataframe = pd.concat([response_dataframe, obj_dataframe], ignore_index=True)
 
-    # Filter out duplicate links
-    response_dataframe.drop_duplicates(subset=["url"], inplace=True)
+        # Validate the URLs
+        for row in response_dataframe['url']:
+            if not validators.url(row):
+                return print("Invalid URL: " + row['url'])
 
-    # Validate the URLs
-    valid_urls = []
-    for url in response_dataframe["url"]:
-        if (validators.url(url) and is_valid_youtube_url(url)):
-            valid_urls.append(url)
-    return valid_urls
+        return response_dataframe
+    except Exception as e:
+        print(f"Error while parsing data: {e}")
+        return pd.DataFrame(columns=['id', 'url', 'game'])
 
-def main(requirements):
+def main():
     """
-    Pulls URLs from the DB that meet the criteria in the requirements argument
+    Pull URLs from the API that meet the criteria specified in the requirements argument.
 
-    Example requirements
-    {
-        "minReviews": 25,
-        "rating": 90,
-    }
+    :param requirements: A dictionary containing the requirements for the URLs
     """
-
     endpoint = args['endpoint']
     params = {
-        "auth_token": args['key'],
-        "requirements": requirements
+        'rating': args['rating'],
+        'minReviews': args['minreviews'],
+        "page": 0
     }
-    params_json = json.dumps(params)
+    headers = {'authorization': args['key'], 'authorization_id': args['id']}
+
     # Make the API request and retrieve the data
-    response = requests.get(endpoint, params=params_json, timeout=10)
-    data = response.json()
-    valid_urls = parse_data(data)
-    # Make output directory if it doesn't exist
-    download_dir = args['output']
-    if not os.path.exists(download_dir):
-        os.makedirs(download_dir)
-    # Save the downloaded links to a file
-        valid_urls_df = pd.DataFrame(valid_urls)
-        valid_urls_df.to_csv((os.path.join(download_dir + "links.csv")),
-                            index=True, columns=["uuid","url"])
+    try:
+        response = requests.get(endpoint, params=params, headers=headers, timeout=10)
+        data = response.json()
+        print(data)
+        total_pages = data["totalPages"]
+
+        valid_urls = pd.DataFrame(columns=['id', 'url', 'game'])
+        for page in range(0, total_pages + 1):  # Query all pages sequentially
```

Comment:
Maybe add an indicator for current page here, since script seems to appear frozen for a long time like:
`print(f"Requesting page {page}")`